playbooks/openstack-deploy_caasp.yml

```diff
@@ -183,18 +183,31 @@
 
         - meta: refresh_inventory
 
-        # Waiting for port is not enough, as the port might be up (systemd)
-        # while ssh is not ready. Wait for connection is not good either
-        # because the key is unknown.
-        - name: Wait for a while for SSH to be properly booted up
+        - name: Wait for a SSH to be properly started
           wait_for:
-            timeout: 120
+            host: "{{ hostvars[item]['ansible_host'] }}"
+            port: 22
+            search_regex: OpenSSH
+            state: started
+            delay: 10
+            timeout: 180
+          loop: "{{ groups['caasp-admin'] + groups['caasp-masters'] + groups['caasp-workers'] }}"
 
         - name: Get pubkey, and add it to known hosts
-          shell: ssh-keyscan -H {{ hostvars[item]['ansible_host'] | quote }} >> ${HOME}/.ssh/known_hosts
+          command: "ssh-keyscan -H {{ hostvars[item]['ansible_host'] | quote }}"
+          register: _caaspsshkeys
+          retries: 10
+          delay: 10
+          until: _caaspsshkeys.stdout and _caaspsshkeys.stdout.strip()
```

Comment:
shouldnt `_caaspsshkeys.stdout` be enough given than an empty string is falsy in python? Or does ansible understand that empty stdout is still stdout?

Reply:
This is mainly to catch the case when stdout is just whitespaces (not sure if this can happen with ssh-keyscan, but "better safe than sorry" :) ) .